pyenv-win/bin/pyenv.bat

```diff
@@ -1,16 +1,103 @@
 @echo off
-setlocal enableDelayedExpansion
+setlocal
 chcp 1250 >nul
 
-IF EXIST "%~dp0"..\exec.bat (
-   del /F /Q "%~dp0"..\exec.bat >nul
+set "pyenv=cscript //nologo "%~dp0..\libexec\pyenv.vbs""
+
+rem if 'pyenv' called alone, then run pyenv.vbs
+if [%1]==[] (
+  %pyenv%
+  exit /b
 )
 
-call cscript //nologo "%~dp0"..\libexec\pyenv.vbs %*
-IF EXIST "%~dp0"..\exec.bat (
-   call "%~dp0"..\exec.bat
+rem use pyenv.vbs to aid resolving absolute path of "active" version into 'bindir'
+set "bindir="
+for /f %%i in ('%pyenv% version') do call :normalizepath "%~dp0..\versions\%%i" bindir
+
+rem all help implemented as plugin
+if /i [%2]==[--help] goto :plugin
+
+rem let pyenv.vbs handle these
+set commands=rehash global local version vname version-name versions commands shims which whence help --help
+for %%a in (%commands%) do (
+  if /i [%1]==[%%a] (
+    %pyenv% %*
+    exit /b
+  )
 )
-for /f "tokens=1 delims=|" %%A in (""!PYENV_VERSION!"") do (
-   endLocal
-   set "PYENV_VERSION=%%~A"
+
+rem jump to plugin or fall to exec
+if /i not [%1]==[exec] goto :plugin
+
+
+rem ====================================================================================
+:exec
+
+echo %bindir%
+if not exist "%bindir%" (
+  echo No global python version has been set yet. Please set the global version.
 )
+
+shift
+
+for /f "tokens=1,2 delims=/" %%i in ("%1") do set "exepath=%%i" & set "exe=%%j"
+if [%exe%]==[] (set "exe=%exepath%") else (set "exe=%bindir%\%exepath%\%exe%")
+set "exe=cmd /c "%exe%""
+goto :run
+
+
+
+
+rem ====================================================================================
+:plugin
+
+set "exe=%~dp0..\libexec\pyenv-%1"
+call :normalizepath %exe% exe
+
+if exist "%exe%.bat" (
+  set "exe=endlocal & call "%exe%.bat""
+
+) else if exist "%exe.cmd%" (
+  set "exe=call "%exe%.cmd""
+
+) else if exist "%exe%.vbs" (
+  set "exe=cscript //nologo "%exe%.vbs""
+
+) else (
+  echo pyenv: no such command '%1'
+  exit /b
+)
+
+
+
+rem ====================================================================================
+:run
+
+rem update PATH to active version
+set "path=%bindir%;%bindir%\Scripts;%path%"
+
+rem copy params to program precisely, preserving double-quotes and percents.
+rem this is the main fix for how 'exec' processing in pyenv.vbs did not
+rem correctly handle (get passed) params with percent chars in them
+set "params="
+:paramloop
+if not [%2]==[] (
+  set params=%params% %2
```

Comment:
```suggestion
  set "params=%params% %2"
```

Reply:
I can't recall the strange edge case I ran into, that led to always wrapping `set` arguments in double quotes, but it was likely related to spaces and/or % expansion and/or quoting behaving non-intuitively.

In this particular case wrapping has no effect, as the the passed parameters (probably better to say passed arguments, but I digress), have already been delimited by spaces. However, I will change for consistency.